apps/api-documenter/src/documenters/MarkdownDocumenter.ts

```diff
@@ -347,6 +349,28 @@ export class MarkdownDocumenter {
         output.appendNode(extendsParagraph);
       }
     }
+
+    if (apiItem instanceof ApiTypeAlias) {
+      const refs: ExcerptToken[] = apiItem.excerptTokens.filter(
```

Comment:
There are a couple interesting edge cases for this:

1. The `build-tests/api-documenter-test/etc/markdown/api-documenter-test.exampletypealias.md` file gets rendered without a hyperlink:

```md
<b>Signature:</b>

```typescript
export declare type ExampleTypeAlias = Promise<boolean>;
```
<b>References:</b> Promise
```
I think this is because `Promise` does have a canonical reference but did not get mapped.  We probably should exclude it in that case.

2. If a type is referenced twice, it does not get deduplicated. For example `type X = Y | typeof Y`.

Reply:
1. Implemented a fix to check if the type actually resolves to another locally declared type before adding a reference.
2. Implemented a simple duplicate checker based on text (please let me know if we should use some other criteria here).

Added test cases for both scenarios.